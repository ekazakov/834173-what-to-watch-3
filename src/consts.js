import PropTypes from "prop-types";

export const VIDEO_TIMER_HOVER = 1000;

export const SAME_FILMS_COUNT = 4;

export const SHOWN_FILMS_DEFAULT = 8;

export const Genres = {
  ALL: `All-genres`,
  HORROR: `horrors`,
};

export const TextCommentLength = {
  MIN: 40,
  MAX: 400,
};

export const MINUTES_IN_HOUR = 60;

export const filmProps = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  posterBig: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  background: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  video: PropTypes.string.isRequired,
  preview: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  score: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(PropTypes.string).isRequired,
  duration: PropTypes.number.isRequired,
  year: PropTypes.number.isRequired,
  favorite: PropTypes.bool.isRequired,
});

export const commentProps = PropTypes.shape({
  id: PropTypes.number.isRequired,
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  rating: PropTypes.number.isRequired,
  comment: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
});

export const filmsProps = PropTypes.arrayOf(filmProps).isRequired;
export const commentsProps = PropTypes.arrayOf(commentProps).isRequired;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const ServerStatus = {
  AVAILABLE: `AVAILABLE`,
  NO_AVAILABLE: `NO_AVAILABLE`,
};

export const TabsName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITE: `/mylist`,
  FILM: `/films`,
  REVIEW: `/review`,
  PLAYER: `/player`,
};

export const FavoriteStatus = {
  ADD: 1,
  DELETE: 0,
};
