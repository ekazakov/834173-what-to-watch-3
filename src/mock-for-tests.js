export const films = [
  {
    id: 1,
    name: `Harry Potter and the Goblet of Fire`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false,
  },
  {
    id: 2,
    name: `EuroTrip`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false,
  },
  {
    id: 3,
    name: `The Autopsy of Jane Doe`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false
  },
  {
    id: 4,
    name: `The Notebook`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false
  },
  {
    id: 5,
    name: `Carri`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false
  },
];

export const comments = [
  {
    id: 0,
    user: {
      id: 0,
      name: `Kate`,
    },
    rating: 7.6,
    comment: `OMG`,
    date: `122`,
  },
  {
    id: 1,
    user: {
      id: 0,
      name: `David`,
    },
    rating: 7.6,
    comment: `OMG`,
    date: `122`,
  },
];
