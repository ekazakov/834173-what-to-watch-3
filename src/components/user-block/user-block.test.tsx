import * as React from "react";
import * as renderer from "react-test-renderer";
import UserBlock from "./user-block";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {AuthorizationStatus} from "../../consts";
import {films, comments} from "../../mock-for-tests";
import {MemoryRouter} from "react-router-dom";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

describe(`Should UserBlock render correctly`, () => {
  it(`Should render for no-authorization user`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        films,
        comments,
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
      },
    });

    const tree = renderer.create(
        <MemoryRouter>
          <Provider store={store}>
            <UserBlock/>
          </Provider>
        </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should render for authorization user`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        films,
        comments,
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    });

    const tree = renderer.create(
        <MemoryRouter>
          <Provider store={store}>
            <UserBlock/>
          </Provider>
        </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
