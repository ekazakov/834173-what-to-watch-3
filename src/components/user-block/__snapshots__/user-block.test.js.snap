// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly Should render for authorization user 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <a
      href="/myList"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        src="img/avatar.jpg"
        width="63"
      />
    </a>
  </div>
</div>
`;

exports[`Should UserBlock render correctly Should render for no-authorization user 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
